### PR TITLE
Fixes failed signature validation

### LINKED ISSUE #321: User is randomly kicked off out of auth session
Details to be completed by @acumarav

### PR DESCRIPTION
**Pull request by acumarav**
Use the central token for communication to CentralAPI
Closes #321

### CODE REVIEW

File: src/main/java/com/odysseusinc/arachne/datanode/controller/AuthController.java [hunk 1]
```diff
@@ -91,52 +89,40 @@
...
-                .orElseThrow(() -> new AuthException("user not registered"));
-        Map<String, String> header = new HashMap<>();
-        header.put(this.tokenHeader, token);
-        String centralToken = centralClient.refreshToken(header).getResult();
-        if (centralToken == null) {
-            throw new AuthException("central auth error");
-        }
-        userService.setToken(user, centralToken);
-        String notSignedToken = centralToken.substring(0, centralToken.lastIndexOf(".") + 1);
+        User user = extractUser(request);
```
Comment (wivern) [inline src/main/java/com/odysseusinc/arachne/datanode/controller/AuthController.java]: code style

File: src/main/java/com/odysseusinc/arachne/datanode/controller/AuthController.java [hunk 2]
```diff
@@ -228,4 +212,16 @@ public ArachnePasswordInfoDTO getPasswordPolicies() throws URISyntaxException {
 
         return integrationService.getPasswordInfo();
     }
+
+    private User extractUser(HttpServletRequest request) {
+        String currentNodeToken = request.getHeader(this.tokenHeader);
```
Comment (wivern) [inline src/main/java/com/odysseusinc/arachne/datanode/controller/AuthController.java]: code style

File: src/main/java/com/odysseusinc/arachne/datanode/controller/AuthController.java [hunk 3]
```diff
@@ -228,4 +212,16 @@ public ArachnePasswordInfoDTO getPasswordPolicies() throws URISyntaxException {
...
+        return userService.findByUsername(tokenUtils.getUsernameFromToken(currentNodeToken))
+                .orElseThrow(() -> new AuthException("user not registered"));
+    }
+
+    private void validateCentralTokenCreated(String centralToken) {
+        if (centralToken == null) {
```
Comment (wivern) [inline src/main/java/com/odysseusinc/arachne/datanode/controller/AuthController.java]: code style

**Comment by acumarav** [comment]
I discovered that existing code sending requests to CentralAPI with the node token which now causes signature validation exception on CentralAPI. I have changed to use centralToken stored in user.token property for requests to central. Looks like signature validation is not happening now in prod or the same password used to generate both tokens.
see new code:
Map<String, String> header = Collections.singletonMap(this.tokenHeader, currentCentralToken);

File: src/main/java/com/odysseusinc/arachne/datanode/controller/AuthController.java (hunk 2 above)
Comment (acumarav) [reply src/main/java/com/odysseusinc/arachne/datanode/controller/AuthController.java]: done

File: src/main/java/com/odysseusinc/arachne/datanode/controller/AuthController.java (hunk 1 above)
Comment (acumarav) [reply src/main/java/com/odysseusinc/arachne/datanode/controller/AuthController.java]: done

File: src/main/java/com/odysseusinc/arachne/datanode/controller/AuthController.java [hunk 4]
```diff
@@ -91,52 +89,41 @@
...
-        header.put(this.tokenHeader, token);
-        String centralToken = centralClient.refreshToken(header).getResult();
-        if (centralToken == null) {
-            throw new AuthException("central auth error");
-        }
-        userService.setToken(user, centralToken);
-        String notSignedToken = centralToken.substring(0, centralToken.lastIndexOf(".") + 1);
+        User user = extractUser(request);
+        String currentCentralToken = user.getToken();
+        Map<String, String> header = Collections.singletonMap(this.tokenHeader, currentCentralToken);
```
Comment (acumarav) [inline src/main/java/com/odysseusinc/arachne/datanode/controller/AuthController.java]: centralToken should be used for the request to CentralAPI

**Comment by wivern** [comment]
I've tested this. Now my concert is only ability of concurrent logins that possibly makes a security hole.

**Comment by acumarav** [comment]
There is the only possibility of concurrent logins to DataNode. Not possible to be logged in to node and portal at the same time. I am not sure if concurrent logins to node allowed or not but anyway it is exists in production so it is not directly related to the current issue scope.

**Pull request merged**